Build is failing because of some type checks:
> Task type_check_client failed after 70353ms.
[12:52:27 ](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Run_All_Unit_Tests/12330110?expandBuildDeploymentsSection=false&hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&showLog=12330110_10846_10846&logFilter=debug&logView=flowAware)       Output for type_check_client
[12:52:27 ](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Run_All_Unit_Tests/12330110?expandBuildDeploymentsSection=false&hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&showLog=12330110_10847_10847&logFilter=debug&logView=flowAware)    If no output is shown, look for it under the test section for this task.
[12:52:27 ](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Run_All_Unit_Tests/12330110?expandBuildDeploymentsSection=false&hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&showLog=12330110_10848_10848&logFilter=debug&logView=flowAware)    ....STDOUT....
[12:52:27 ](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Run_All_Unit_Tests/12330110?expandBuildDeploymentsSection=false&hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&showLog=12330110_10849_10849&logFilter=debug&logView=flowAware)    client/components/theme-upgrade-modal/index.tsx(217,4): error TS2345: Argument of type 'ExistingReactNode' is not assignable to parameter of type 'string'.
[12:52:27 ](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Run_All_Unit_Tests/12330110?expandBuildDeploymentsSection=false&hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&showLog=12330110_10850_10850&logFilter=debug&logView=flowAware)      Type 'number' is not assignable to type 'string'.
[12:52:27 ](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Run_All_Unit_Tests/12330110?expandBuildDeploymentsSection=false&hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&showLog=12330110_10851_10851&logFilter=debug&logView=flowAware)    client/components/theme-upgrade-modal/index.tsx(311,4): error TS2345: Argument of type 'ExistingReactNode' is not assignable to parameter of type 'string'.
[12:52:27 ](https://teamcity.a8c.com/buildConfiguration/calypso_calypso_WebApp_Run_All_Unit_Tests/12330110?expandBuildDeploymentsSection=false&hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&showLog=12330110_10852_10852&logFilter=debug&logView=flowAware)      Type 'number' is not assignable to type 'string'.

--- a/apps/wpcom-block-editor/src/wpcom/features/live-preview/hooks/use-can-preview-but-need-upgrade.ts
+++ b/apps/wpcom-block-editor/src/wpcom/features/live-preview/hooks/use-can-preview-but-need-upgrade.ts
@@ -136,17 +136,20 @@ export const useCanPreviewButNeedUpgrade = (
 	const [ checkoutTab, setCheckoutTab ] = useState< Window | null >();
 	const [ checkoutStatus, setCheckoutStatus ] = useState< CheckoutStatus >( '' );
 
-	// TODO: This doesn't work on sites with longer length plans, the plan should be the
-	// biannual / trieenial plan.
-	let requiredPlanSlug = '';
-	switch ( previewingTheme.type ) {
-		case WOOCOMMERCE_THEME:
-			requiredPlanSlug = PLAN_BUSINESS;
-		case PREMIUM_THEME:
-			requiredPlanSlug = PLAN_PREMIUM;
-		case PERSONAL_THEME:
-			requiredPlanSlug = PLAN_PERSONAL;
-	}
+	const getRequiredPlanSlug = (): string => {
+		switch ( previewingTheme.type ) {
+			case WOOCOMMERCE_THEME:
+				return PLAN_BUSINESS;
+			case PREMIUM_THEME:
+				return PLAN_PREMIUM;
+			case PERSONAL_THEME:
+				return PLAN_PERSONAL;
+			default:
+				return '';
+		}
+	};
+
+	const requiredPlanSlug = getRequiredPlanSlug();
 
 	const handleCanPreviewButNeedUpgrade = useCallback(
 		( previewingTheme: ReturnType< typeof usePreviewingTheme > ) => {

--- a/client/components/theme-upgrade-modal/index.tsx
+++ b/client/components/theme-upgrade-modal/index.tsx
@@ -214,9 +214,9 @@ export const ThemeUpgradeModal = ( {
 		const planPrice = requiredPlanProduct?.combined_cost_display;
 
 		const planText = getPlanText(
-			premiumPlanName,
+			premiumPlanName as string,
 			requiredPlanProduct?.product_term || '',
-			planPrice
+			planPrice || ''
 		);
 
 		return {
@@ -308,9 +308,9 @@ export const ThemeUpgradeModal = ( {
 		const color = bundleSettings.color;
 		const Icon = bundleSettings.iconComponent;
 		const planText = getPlanText(
-			businessPlanName,
+			businessPlanName as string,
 			requiredPlanProduct?.product_term || '',
-			businessPlanPrice
+			businessPlanPrice || ''
 		);
 
 		return {

--- a/client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx
+++ b/client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx
@@ -949,7 +949,7 @@ function getBusinessPlanByTerm( term: string ) {
 }
 
 function getRequiredPlan( selectedDesign: Design | undefined, currentPlanSlug: string ) {
-	if ( ! selectedDesign || ! selectedDesign?.design_tier ) {
+	if ( ! selectedDesign?.design_tier ) {
 		return;
 	}
 	// Different designs require different plans to unlock them, additionally the terms required can vary.
@@ -962,11 +962,7 @@ function getRequiredPlan( selectedDesign: Design | undefined, currentPlanSlug: s
 	let requiredTerm;
 	if ( ! currentPlanSlug || isFreePlan( currentPlanSlug ) ) {
 		// Marketplace themes require upgrading to a monthly business plan or higher, everything else requires an annual plan.
-		if ( selectedDesign?.is_externally_managed ) {
-			requiredTerm = TERM_MONTHLY;
-		} else {
-			requiredTerm = TERM_ANNUALLY;
-		}
+		requiredTerm = selectedDesign?.is_externally_managed ? TERM_MONTHLY : TERM_ANNUALLY;
 	} else {
 		requiredTerm = getPlan( currentPlanSlug )?.term || '';
 	}

